src/main/java/lesson2/homework/PerformanceTester.java

```diff
@@ -0,0 +1,148 @@
+package lesson2.homework;
+
+import java.util.ArrayList;
+import java.util.LinkedList;
+import java.util.Random;
+import java.util.Set;
+import java.util.concurrent.TimeUnit;
+
+public class PerformanceTester {
+
+  ArrayList<Long> results = new ArrayList<>();
+
+  private long start;
+  private long end;
+
+  private long startExternal;
+  private long pointExternal;
+  private long endExternal;
+
+  public void measureStart() {
+    startExternal = System.nanoTime();
+  }
+
+  public void measurePoint() {
+    pointExternal = System.nanoTime();
+  }
+
+  public void measureFinish() {
+    endExternal = System.nanoTime();
+    long result = endExternal - startExternal;
+    System.out.println();
+    System.out.println("Test general result is: " + TimeUnit.NANOSECONDS.toMillis(result) + "ms");
+    if (pointExternal != 0) {
+      long resultPointToStart = pointExternal - startExternal;
+      long resultPointToEnd = endExternal - pointExternal;
+      System.out.println();
+      System.out.println(
+          "Test result Point -> Start is: " + TimeUnit.NANOSECONDS.toMillis(resultPointToStart));
+      System.out.println(
+          "Test result Point -> End is: " + TimeUnit.NANOSECONDS.toMillis(resultPointToEnd));
+      pointExternal = 0;
+      System.out.println();
+    }
+  }
+
+  public void measureFinish(String message) {
+    endExternal = System.nanoTime();
+    long result = endExternal - startExternal;
+    System.out.println();
+    System.out.println(
+        "Test general result is: " + TimeUnit.NANOSECONDS.toMillis(result) + "ms, " + message);
+    if (pointExternal != 0) {
+      long resultPointToStart = pointExternal - startExternal;
+      long resultPointToEnd = endExternal - pointExternal;
+      System.out.println();
+      System.out.println(
+          "Test result Point -> Start is: " + TimeUnit.NANOSECONDS.toMillis(resultPointToStart));
+      System.out.println(
+          "Test result Point -> End is: " + TimeUnit.NANOSECONDS.toMillis(resultPointToEnd));
+      pointExternal = 0;
+    }
+  }
+
+
+  public void addElements(ArrayList list, int iterations, int times) {
+    beforeTest();
+
+    for (int i = 1; i <= times; i++) {
+      list.clear();
+      int x = 0;
+      start = System.nanoTime();
+      while (x <= iterations) {
+        list.add(getRandomIntValue());
+        x++;
+      }
+      end = System.nanoTime();
+      saveResult(getResult());
+    }
+    finalResult(times);
+    System.out.println(list.size());
+  }
+
+  public void addElements(LinkedList<Integer> list, int iterations, int times) {
```

Comment:
лучше оставить один универсальный метод. Дублирование кода всегда плохо, нужно поддерживать больше кода
```suggestion
  public void addElements(Collection<Integer> list, int iterations, int times) {
```

Reply:
я хотел так сделать, но у меня не получилось. Сейчас проверил - получилось.. магия).
Спасибо. Для Мар нужно будет свой метод все равно писать.

Reply:
public void addElements(Collection<?> list, int iterations, int times) {
наверное даже так можно. какая разница какие типы складывать для теста